src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -393,7 +451,338 @@ public virtual bool TryInsertExpansion(int startPositionInSubjectBuffer, int end
+            {
+                // Full method call completion is not enabled
+                return false;
+            }
+
+            var symbols = ThreadingContext.JoinableTaskFactory.Run(() =>
+            {
+                var caretPosition = SubjectBuffer.CurrentSnapshot.GetPoint(endPositionInSubjectBuffer);
+                return GetSymbolsAsync(caretPosition, cancellationToken);
+            });
+
+            if (symbols.OfType<IMethodSymbol>().Any())
+            {
+                XNamespace snippetNamespace = "http://schemas.microsoft.com/VisualStudio/2005/CodeSnippet";
+
+                var methodName = dataBufferSpan.GetText();
+
+                var template = $"{methodName}($placeholder$)$end$";
+
+                var snippet = new XDocument(
+                    new XDeclaration("1.0", "utf-8", null),
+                    new XElement(
+                        snippetNamespace + "CodeSnippets",
+                        new XElement(
+                            snippetNamespace + "CodeSnippet",
+                            new XAttribute(snippetNamespace + "Format", "1.0.0"),
+                            new XElement(
+                                snippetNamespace + "Header",
+                                new XElement(
+                                    snippetNamespace + "SnippetTypes",
+                                    new XElement(snippetNamespace + "SnippetType", new XText("Expansion"))),
+                                new XElement(snippetNamespace + "Title", new XText(methodName)),
+                                new XElement(snippetNamespace + "Author", "Microsoft"),
+                                new XElement(snippetNamespace + "Description"),
+                                new XElement(snippetNamespace + "HelpUrl"),
+                                new XElement(snippetNamespace + "Shortcut", methodName)),
+                            new XElement(
+                                snippetNamespace + "Snippet",
+                                new XElement(
+                                    snippetNamespace + "Declarations",
+                                    new XElement(
+                                        snippetNamespace + "Literal",
+                                        new XAttribute(snippetNamespace + "Editable", "true"),
+                                        new XElement(snippetNamespace + "ID", new XText("placeholder")),
+                                        new XElement(snippetNamespace + "ToolTip", new XText("")),
+                                        new XElement(snippetNamespace + "Default", new XText("")),
+                                        new XElement(snippetNamespace + "Function"))),
+                                new XElement(
+                                    snippetNamespace + "Code",
+                                    new XAttribute(snippetNamespace + "Language", "csharp"),
+                                    new XCData(template))))));
```

Comment:
plzdoc.  :)

Reply:
also, consider extracting this to helper to keep this metohd itself cleaner and clearer.

Reply:
⏱️ Extracted and documented a shared helper.